examples/ecs/fetch.rs

```diff
@@ -0,0 +1,93 @@
+use bevy::ecs::component::Component;
+use bevy::{
+    ecs::query::{Fetch, FilterFetch},
+    prelude::*,
+};
+use std::marker::PhantomData;
+
+fn main() {
+    App::new()
+        .add_startup_system(spawn)
+        .add_system(print_nums)
+        .add_system(print_nums_readonly)
+        .run();
+}
+
+#[derive(Fetch, Debug)]
+struct NumQuery<'w, T: Component, P: Component> {
+    entity: Entity,
+    u: UNumQuery<'w>,
+    i: INumQuery<'w>,
+    generic: MutGenericQuery<'w, T, P>,
+    #[filter(NumQueryFilter<T, P>)]
+    filter: bool,
+}
+
+// If you want to declare a read-only query that uses nested `Fetch` structs, you need to
+// specify `readonly` attribute for the corresponding fields. This will generate static assertions
+// that those members implement `ReadOnlyFetch`.
+#[derive(Fetch, Debug)]
+struct ReadOnlyNumQuery<'w, T: Component, P: Component> {
+    entity: Entity,
+    #[readonly]
+    u: UNumQuery<'w>,
+    #[readonly]
+    generic: GenericQuery<'w, T, P>,
+    #[filter(NumQueryFilter<T, P>)]
+    filter: bool,
+}
+
+#[derive(Fetch, Debug)]
+struct UNumQuery<'w> {
+    u_16: &'w u16,
```

Comment:
Using built-in types like this is a) unclear b) bad practice c) about to break with #2254. 

We should use simple `ComponentA` wrapper types instead.